src/main/java/dan200/computercraft/client/gui/widgets/WidgetTerminal.java

```diff
@@ -221,37 +227,63 @@ public void handleMouseInput( int mouseX, int mouseY )
                 charX = Math.min( Math.max( charX, 0 ), term.getWidth() - 1 );
                 charY = Math.min( Math.max( charY, 0 ), term.getHeight() - 1 );
 
-                if( m_lastClickButton >= 0 && !Mouse.isButtonDown( m_lastClickButton ) )
-                {
-                    if( m_focus ) computer.mouseUp( m_lastClickButton + 1, charX + 1, charY + 1 );
-                    m_lastClickButton = -1;
-                }
+                handleMouseClick( computer, charX, charY );
+                handleMouseWheel( computer, charX, charY );
+                handleMouseMove( charX, charY );
+            }
+        }
+        else // The mouse has moved out of the terminal, send a -1, -1 mouse_move event
+        {
+            handleMouseMove( -1, -1 );
```

Comment:
Are we sure this is a good idea, instead of sending _no_ position (so that the code explodes with, say, `attempt to perform arithmetic on nil`) or a different event for losing focus entirely?

Reply:
My biggest concern with sending `nil` is backwards compatibility with programs that perform blanket handling of all mouse events, but `-1` could have a similar outcome.

Reply:
I'd be inclined to send nil<sup>1</sup>, or have an entirely separate event. I don't know, both are gross. If someone's doing `event:find("^mouse_")` then we're already breaking their code due to missing the "button" argument.

Also, does this send the event every frame, or just the first one it moves outside?

1: Well, send 0 or -1, but convert it to `nil` on the server.